src/tests/simple_tests/uma.rs

```diff
@@ -0,0 +1,39 @@
+#[cfg(test)]
+mod tests {
+    use crate::tests::simple_tests::{asm_tests::run_asm_based_test, Options};
+
+    #[test_log::test]
+    fn test_uma_reads_and_writes() {
+        run_asm_based_test(
+            "src/tests/simple_tests/testdata/uma/reads_and_writes",
+            &[800000],
+            Options {
+                cycle_limit: 100,
+                cycles_per_vm_snapshot: 1,
+                ..Default::default()
+            },
+        )
+        .unwrap();
+    }
+
+    #[test_log::test]
+    fn test_uma_static_reads() {
+        let error = run_asm_based_test(
+            "src/tests/simple_tests/testdata/uma/static_reads",
+            &[800000],
+            Options {
+                cycle_limit: 100,
+                cycles_per_vm_snapshot: 1,
+                ..Default::default()
+            },
+        )
+        .err()
+        .expect("Expected this test to fail");
```

Comment:
I don't think this test will attempt to compare out-of-circuit and in-circuit. It will return after panic in out-of-circuit VM

Reply:
you're absolutely right. good catch.